cl/citations/tasks.py

```diff
@@ -114,28 +119,58 @@ def find_citations_and_parantheticals_for_recap_documents(
 def find_citations_and_parentheticals_for_opinion_by_pks(
     self,
     opinion_pks: List[int],
+    disconnect_pg_signals: bool = False,
 ) -> None:
     """Find citations and authored parentheticals for search.Opinion objects.
 
     :param opinion_pks: An iterable of search.Opinion PKs
+    :param disconnect_pg_signals: True if ParentheticalGroup post_save and
+        post_delete signals should be disconnected; useful in batch jobs
+        from the `find_citations` command
+
     :return: None
     """
     opinions: QuerySet[Opinion, Opinion] = Opinion.objects.filter(
         pk__in=opinion_pks
     )
-    for opinion in opinions:
+    # delivery_info does not exist in test environment
+    children_queue = (self.request.delivery_info or {}).get(
+        "routing_key", settings.CELERY_ETL_TASK_QUEUE
+    )
+
+    for index, opinion in enumerate(opinions):
+        if disconnect_pg_signals:
+            disconnect_parenthetical_group_signals()
+
```

Comment:
I think we can avoid connecting and disconnecting the PA signals on every iteration. If we move it outside the loop, we can disconnect and reconnect only once per task? The behavior should remain the same, so any PA save or delete operation executed by the same Celery worker running the task won't trigger PA indexing signals.